### PR TITLE
IDEX-3166: Adopt new style for notifications popups and notifications

### PR DESCRIPTION
**Pull request by vzhukovskii**
@vparfonov 

Screencast attached to the issue https://jira.codenvycorp.com/browse/IDEX-3166

### CODE REVIEW

File: commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/MathUtil.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+import static java.lang.Math.pow;
+
+/**
+ * Provides utility methods that helps math computations.
+ *
+ * @author Vlad Zhukovskiy
+ */
+public class MathUtil {
+    /**
```
Comment (skabashnyuk) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/MathUtil.java]: Is this logic related to IDE somehow?

Comment (vzhukovskii) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/MathUtil.java]: This logic is a part of animation flow.

**Comment by evoevodin** [comment]
In the cases of util classes it is always good to defined `private` default constructor + make the class final.

File: commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/DebugUtils.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+     * @return generated id
+     */
+    public static String getDebugId(String prefix, IDProvider provider) {
+        StringBuilder sb = new StringBuilder();
+
+        if (prefix != null & !prefix.isEmpty()) {
+            sb.append(prefix).append('-');
+        }
+
+        sb.append(provider.getId());
```
Comment (evoevodin) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/DebugUtils.java]: What if provider is `null` ?

File: commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/DebugUtils.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+     * @param prefix
+     *         prefix
+     * @param provider
+     *         object that provides string id property
+     * @return generated id
+     */
+    public static String getDebugId(String prefix, IDProvider provider) {
+        StringBuilder sb = new StringBuilder();
+
+        if (prefix != null & !prefix.isEmpty()) {
```
Comment (evoevodin) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/DebugUtils.java]: You get NPE here, please be careful with binary operators

Comment (vzhukovskii) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/DebugUtils.java]: It was typo error, thx, fixed

File: ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java [hunk 4]
```diff
@@ -33,462 +33,99 @@
...
-     *
-     * @param message
-     *         notification's message
-     * @param status
-     *         notification's status
-     */
-    public Notification(@NotNull String message, @NotNull Status status) {
-        this(message, status, null, null);
+    public Notification(@Nullable ProjectConfigDto project,
+                        @NotNull String title,
```
Comment (evoevodin) [inline ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java]: What if `title` is `null` ?

File: ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java [hunk 5]
```diff
@@ -33,462 +33,99 @@
...
+                        @NotNull String message) {
+        this(project, title, message, null);
```
Comment (evoevodin) [inline ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java]: This code looks strange for me as you defined listener as `NotNull` but you are using `null`  here
```java
public Notification(@Nullable ProjectConfigDto project,
                    @NotNull String title,
                    @NotNull String message,
                    @NotNull NotificationListener listener) {
       id = Document.get().createUniqueId();
       observers = new ArrayList<>();
```

File: ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java [hunk 6]
```diff
@@ -33,462 +33,99 @@
...
-     *         notification's message
-     * @param status
-     *         notification's status
-     * @param important
-     *         note about important this notification
+     * @return notification event listener
      */
-    public Notification(@NotNull String message, @NotNull Status status, boolean important) {
-        this(message, status, important, null, null);
+    public NotificationListener getListener() {
```
Comment (evoevodin) [inline ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java]: Is it possible to get `null` from this method? Please define a contract

File: ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java [hunk 7]
```diff
@@ -33,462 +33,99 @@
...
-     *         note about important this notification
-     * @param closeHandler
-     *         delegate that provides some actions when closing notification
-     */
-    public Notification(@NotNull String message, @NotNull Status status, boolean important,
-                        @NotNull CloseNotificationHandler closeHandler) {
-        this(message, INFO, status, UNREAD, new Date(), important, null, closeHandler);
+    protected void setUnreadState() {
+        //lets collect all calls instead of continuously calling update observers
+        setUnreadStateTask.delay(200);
```
Comment (evoevodin) [inline ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java]: Why its 200 ?
It seems that it should be moved to kind of _constants_ place and explained

File: ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java (hunk 4 above)
Comment (vzhukovskii) [inline ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java]: Notification title will be "null".

File: ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/NotificationObserver.java [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.api.notification;
+
+/** Notification observer. */
```
Comment (evoevodin) [inline ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/NotificationObserver.java]: author ?

File: ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java (hunk 4 above)
Comment (vzhukovskii) [inline ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java]: Added correct annotations

Comment (evoevodin) [inline ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java]: If you mark parameter as `NotNull` you must define the contarct for this parameter, in other words you must define what i should expect when i pass `null` to this method. You didn't define it, see [example](https://github.com/codenvy/che-core/blob/4.0/platform-api/che-core-api-core/src/main/java/org/eclipse/che/api/core/rest/HttpJsonRequest.java#L71-L74).

What is more important is to define this contract for interfaces, as implementator won't know what  to do in the case of nulls.

File: commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/DebugUtils.java [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+        if (prefix != null && !prefix.isEmpty()) {
+            sb.append(prefix).append('-');
+        }
+
+        sb.append(provider.getId());
+
+        return sb.toString();
+    }
+
+    private DebugUtils() {
```
Comment (evoevodin) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/DebugUtils.java]: Do you follow `che-new` [codestyle](https://github.com/codenvy/codenvy-dev-resources/blob/master/che-codestyle/src/main/resources/che-codestyle-intellij-new.xml) ?

File: ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java (hunk 4 above)
Comment (vzhukovskii) [inline ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java]: Not every time. Its true for the server side, but for the client side behaviour can be unexpected. So its impossible to define all possible behiours.

Comment (vzhukovskii) [inline ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java]: I added javadoc for the setter and I think that's enought

File: commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/DebugUtils.java (hunk 9 above)
Comment (vzhukovskii) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/DebugUtils.java]: old one

File: ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java (hunk 7 above)
Comment (vzhukovskii) [inline ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java]: It's an experiment value

File: commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/DebugUtils.java [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+import org.eclipse.che.commons.annotation.Nullable;
+
+import javax.validation.constraints.NotNull;
+
+/**
+ * Provides debug id for the class that implements {@link IDProvider}
+ *
+ * @author Vlad Zhukovskiy
+ */
+public final class DebugUtils {
```
Comment (vitaliy-guliy) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/DebugUtils.java]: Why do we need this class?

File: commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/IDProvider.java [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.util;
+
+/**
+ * Provides ID for the object.
+ *
+ * @author Vlad Zhukovskiy
+ */
+public interface IDProvider {
```
Comment (vitaliy-guliy) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/IDProvider.java]: Why do we need this class? Only Notification implements it.

File: commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/MathUtils.java [hunk 12]
```diff
@@ -1,24 +1,50 @@
...
+     *         coordinate for the x1
+     * @param x2
+     *         coordinate for the x2
+     * @param x3
+     *         coordinate for the x3
+     * @param x4
+     *         coordinate for the x4
+     * @return coordinate for the point
+     */
+    public static double getCubicBezier(double t, double x1, double x2, double x3, double x4) {
```
Comment (vitaliy-guliy) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/MathUtils.java]: The method is called only once. So, we can delete the class and move the method inside NotificationPopup.java

File: commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/MathUtils.java [hunk 13]
```diff
@@ -1,24 +1,50 @@
...
+        return pow((1. - t), 3.) * x1 + 3. * pow((1. - t), 2.) * t * x2 + 3. * (1. - t) * pow(t, 2.) * x3 + pow(t, 3.) * x4;
```
Comment (vitaliy-guliy) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/MathUtils.java]: It's difficult to understand

File: commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/DebugUtils.java (hunk 10 above)
Comment (vzhukovskii) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/DebugUtils.java]: Because I saw many pieces of code that perform some bycicles...

Its a proposal to unify all this work. 

note: notification - its not a gwt component its a simple data object
```
        SimplePanel iconWrapper = new SimplePanel();
        iconWrapper.ensureDebugId(DebugUtils.getDebugId("some-prefix", notification));
        return iconWrapper;
```

File: commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/IDProvider.java (hunk 11 above)
Comment (vzhukovskii) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/IDProvider.java]: This is a proposal to unify mechanism of providing debug id for the gwt components based on their data objects

File: commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/MathUtils.java (hunk 12 above)
Comment (vzhukovskii) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/MathUtils.java]: Disagree, lets all mathematical methods will be placed in one place and not layed on whole code

File: commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/MathUtils.java (hunk 13 above)
Comment (vzhukovskii) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/MathUtils.java]: Its default notation of cubic bezier function which is provided on the wiki I mentioned in the javadoc

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/notification/NotificationPopup.java [hunk 14]
```diff
@@ -0,0 +1,374 @@
...
+            update();
+
+            new Animation() {
+
+                /** {@inheritDoc} */
+                @Override
+                protected void onUpdate(double progress) {
+                    getElement().getStyle().setOpacity(progress);
+                    getElement().getStyle().setProperty("transform", "scale(" + (2 - progress) + ',' + (2 - progress) + ");");
+                    getElement().getStyle().setProperty("filter", "blur(" + (2 - progress * 2) + "px);");
```
Comment (vitaliy-guliy) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/notification/NotificationPopup.java]: Using css animation will optimize the code and speed up the app.

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/notification/NotificationPopup.java [hunk 15]
```diff
@@ -0,0 +1,374 @@
...
+    public void removeFromParent() {
+        if (isOrWasAttached()) {
+            new Animation() {
+                double offsetWidth = getElement().getOffsetWidth();
+
+                /** {@inheritDoc} */
+                @Override
+                protected void onUpdate(double progress) {
+                    getElement().getStyle().setMarginLeft(offsetWidth * getCubicBezier(progress, 0, .05, 0, 1), PX);
+                    getElement().getStyle().setOpacity(1 - progress);
```
Comment (vitaliy-guliy) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/notification/NotificationPopup.java]: Using CSS animations is the best way.

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/notification/NotificationPopup.java (hunk 14 above)
Comment (vzhukovskii) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/notification/NotificationPopup.java]: I had some troubles with applying pure css animations to popups, so I decided to do this in simple way, via classic animations

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/notification/NotificationPopup.java [hunk 16]
```diff
@@ -0,0 +1,374 @@
...
+
+        this.notification = notification;
+        this.resources = resources;
+        this.delegate = delegate;
+
+        setStyleName(resources.notificationCss().notificationPopupPanel());
+
+        notification.addObserver(this);
+
+        FlowPanel contentWrapper = new FlowPanel("div");
```
Comment (vitaliy-guliy) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/notification/NotificationPopup.java]: just do new FlowPanel()

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/notification/NotificationPopup.java (hunk 15 above)
Comment (vzhukovskii) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/notification/NotificationPopup.java]: css animation doesn't work when we physically remove popup from DOM model

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/notification/NotificationPopup.java (hunk 16 above)
Comment (vzhukovskii) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/notification/NotificationPopup.java]: done

File: ide/che-core-ide-app/src/main/resources/org/eclipse/che/ide/notification/notification.css [hunk 17]
```diff
@@ -8,132 +8,206 @@
...
-    padding: 10px;
+.notificationIconWrapper svg[name="progress"] {
+    fill: #9b9b9b;
 }
 
-.closePopupIcon:hover {
-    opacity: 1;
+.notificationContentWrapper{
+    display: inline-block;
+    width: literal("calc(100% - 75px)");
```
Comment (vitaliy-guliy) [inline ide/che-core-ide-app/src/main/resources/org/eclipse/che/ide/notification/notification.css]: Not sure but this code may not work on some portable devices.

Comment (vzhukovskii) [inline ide/che-core-ide-app/src/main/resources/org/eclipse/che/ide/notification/notification.css]: we don't support portable devices.

File: commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/DebugUtils.java (hunk 10 above)
Comment (vitaliy-guliy) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/DebugUtils.java]: It looks the following is used only for concatenating strings.

File: commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/MathUtils.java (hunk 12 above)
Comment (ashumilova) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/MathUtils.java]: what other math calculations can be added here?

File: ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java [hunk 18]
```diff
@@ -33,462 +33,109 @@
...
+
+    protected String                     id;
+    protected long                       time;
+    private   List<NotificationObserver> observers;
+
+    protected String           title;
+    protected String           message;
+    protected ReadState        state;
+    protected ProjectConfigDto project;
+    protected DelayedTask setUnreadStateTask = new DelayedTask() {
```
Comment (ashumilova) [inline ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java]: need to add alignment here

File: commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/MathUtils.java (hunk 12 above)
Comment (azatsarynnyy) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/MathUtils.java]: I think that it's unnecessarily to add new utility class with one method that used only in one place.

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/DeleteItemAction.java [hunk 19]
```diff
@@ -173,7 +173,7 @@ protected void onSuccess(final Status result) {
 
                     @Override
                     protected void onFailure(Throwable exception) {
-                        notificationManager.showError(exception.getMessage());
+                        notificationManager.notify("Git", exception.getMessage());
```
Comment (ashumilova) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/DeleteItemAction.java]: why Git?

Comment (sleshchenko) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/DeleteItemAction.java]: Why we should send String as parameter? It can produce many problems with copy/paste, refactoring etc. What benefits we get from this?

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/newresource/AbstractNewResourceAction.java [hunk 20]
```diff
@@ -171,12 +171,13 @@ public void accepted() {
                                  new RequestCallback<Void>() {
                                      @Override
                                      protected void onSuccess(Void result) {
-                                         notificationManager.showInfo(localizationConstant.actionNewFileAddToIndexNotification(fileName));
+                                         notificationManager
+                                                 .notify("Git", localizationConstant.actionNewFileAddToIndexNotification(fileName));
```
Comment (ashumilova) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/newresource/AbstractNewResourceAction.java]: why Git?

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/newresource/AbstractNewResourceAction.java [hunk 21]
```diff
@@ -171,12 +171,13 @@ public void accepted() {
...
                                      }
 
                                      @Override
                                      protected void onFailure(Throwable exception) {
-                                         notificationManager.showError(exception.getMessage());
+                                         notificationManager.notify("Git", exception.getMessage());
```
Comment (ashumilova) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/newresource/AbstractNewResourceAction.java]: why Git?

File: ide/che-core-ide-app/src/main/resources/org/eclipse/che/ide/notification/notification.css [hunk 22]
```diff
@@ -8,132 +8,206 @@
...
+.notificationPopupIconWrapper {
+    display: inline-block;
+    vertical-align: top;
+    margin: 18px 18px 15px 18px;
+}
+
+.notificationPopupIconWrapper > svg > g,
+.notificationPopupIconWrapper > svg > rect,
+.notificationPopupIconWrapper > svg > path {
+    fill: #FFF;
```
Comment (ashumilova) [inline ide/che-core-ide-app/src/main/resources/org/eclipse/che/ide/notification/notification.css]: use constant here

File: ide/che-core-ide-app/src/main/resources/org/eclipse/che/ide/notification/notification.css [hunk 23]
```diff
@@ -8,132 +8,206 @@
...
-    margin-top: 3px;
-    margin-left: 3px;
-    padding: 0px 3px;
-    color: tabBorderColor;
-    background-color: mainFontColor;
-    line-height: 11px;
-    vertical-align: middle;
+.notificationPopupPanel {
+    border-radius: 5px;
+    box-shadow: 0 0 10px rgba(0,0,0,0.6);
```
Comment (ashumilova) [inline ide/che-core-ide-app/src/main/resources/org/eclipse/che/ide/notification/notification.css]: use constant here

File: ide/che-core-ide-app/src/main/resources/org/eclipse/che/ide/notification/notification.css [hunk 24]
```diff
@@ -8,132 +8,206 @@
...
-    fill: successEventColor;
+.notificationStatusSuccess {
+    background-color: #31b993;
 }
 
-.warning {
-    color: #cb772f !important;
-    fill: #cb772f;
+.notificationStatusFail {
+    background-color: #c34d4d;
```
Comment (ashumilova) [inline ide/che-core-ide-app/src/main/resources/org/eclipse/che/ide/notification/notification.css]: use constant here

File: ide/che-core-ide-app/src/main/resources/org/eclipse/che/ide/notification/notification.css [hunk 25]
```diff
@@ -8,132 +8,206 @@
...
-    color: errorEventColor !important;
-    fill: errorEventColor !important;
+.notificationStatusProgress {
+    background-color: #9b9b9b;
 }
 
-.success {
-    fill: successEventColor;
+.notificationStatusSuccess {
+    background-color: #31b993;
```
Comment (ashumilova) [inline ide/che-core-ide-app/src/main/resources/org/eclipse/che/ide/notification/notification.css]: use constant here

File: ide/che-core-ide-app/src/main/resources/org/eclipse/che/ide/notification/notification.css [hunk 26]
```diff
@@ -8,132 +8,206 @@
...
-    margin-right: 25px !important;
+.notificationCloseButtonWrapper svg:hover g {
+    fill: minimizeIconHoverColor;
 }
 
-.error {
-    color: errorEventColor !important;
-    fill: errorEventColor !important;
+.notificationStatusProgress {
+    background-color: #9b9b9b;
```
Comment (ashumilova) [inline ide/che-core-ide-app/src/main/resources/org/eclipse/che/ide/notification/notification.css]: use constant here

File: ide/che-core-ide-app/src/main/resources/org/eclipse/che/ide/notification/notification.css [hunk 27]
```diff
@@ -8,132 +8,206 @@
...
-    height: 15px;
+.notificationIconWrapper svg {
+    width: 16px;
+    height: 16px;
 }
 
-.notificationItem a, .notificationPopup a {
-    color: linkColor;
+.notificationIconWrapper svg[name="fail"] {
+    fill: #c34d4d;
```
Comment (ashumilova) [inline ide/che-core-ide-app/src/main/resources/org/eclipse/che/ide/notification/notification.css]: use constant here

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/DeleteItemAction.java (hunk 19 above)
Comment (vzhukovskii) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/DeleteItemAction.java]: Because we got failed response from git status operation.

"Git" - its title for the notification.

File: ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java (hunk 18 above)
Comment (vzhukovskii) [inline ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java]: Formatter talks that all fine

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/newresource/AbstractNewResourceAction.java (hunk 20 above)
Comment (vzhukovskii) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/newresource/AbstractNewResourceAction.java]: Because we got failed response from git add operation.

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/newresource/AbstractNewResourceAction.java (hunk 21 above)
Comment (vzhukovskii) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/newresource/AbstractNewResourceAction.java]: Because we got response from git add operation.

File: commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/MathUtils.java (hunk 12 above)
Comment (vzhukovskii) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/MathUtils.java]: If somebody will create another functionality that uses for example this function, he will know that in commons module there is a Math utility class instead of searching for it inside other modules.

Comment (vzhukovskii) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/MathUtils.java]: BTW, MahtUtils class was created before my PR

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/DeleteItemAction.java (hunk 19 above)
Comment (sleshchenko) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/DeleteItemAction.java]: Should we use title of notification from locale?

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/newresource/AbstractNewResourceAction.java (hunk 21 above)
Comment (vzhukovskii) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/newresource/AbstractNewResourceAction.java]: Another question why git operations we perform in che-core?

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/DeleteItemAction.java (hunk 19 above)
Comment (vzhukovskii) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/DeleteItemAction.java]: I don't want to add git dependency from che-plugins only for the single locale constant. We should avoid git operation calls in che-core otherwise.

**Comment by vparfonov** [comment]
ok, but also aware about titles like "Git" and don't commit commented test class

**Comment by vzhukovskii** [comment]
Reverted test removation

File: ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java [hunk 28]
```diff
@@ -33,462 +32,109 @@
...
-     *         note about important this notification
-     * @param openHandler
-     *         delegate that provides some actions when opening notification
-     * @param closeHandler
-     *         delegate that provides some actions when closing notification
-     */
-    public Notification(@NotNull String message, @NotNull Type type, @NotNull Status status, boolean important,
-                        @Nullable OpenNotificationHandler openHandler, @Nullable CloseNotificationHandler closeHandler) {
-        this(message, type, status, UNREAD, new Date(), important, openHandler, closeHandler);
+    public Notification(@Nullable ProjectConfigDto project,
```
Comment (sleshchenko) [inline ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java]: It would be better to put this nullable (optional) parameter to the end of parameters list

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/notification/NotificationResources.java [hunk 29]
```diff
@@ -12,66 +12,66 @@
...
+    @Source("fail.svg")
+    SVGResource fail();
 
-    @Source("warning.svg")
-    SVGResource warning();
-
-    @Source("close-popup.svg")
-    SVGResource closePopup();
+    @Source("progress.svg")
+    SVGResource progress();
```
Comment (vitaliy-guliy) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/notification/NotificationResources.java]: Progress, success and error images are used in plugin-openshift. Did you check it?
